tests/csapi/apidoc_content_test.go

```diff
@@ -0,0 +1,63 @@
+package csapi_tests
+
+import (
+	"io/ioutil"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+)
+
+var contentID *url.URL
+var contentText = "Here is the content I am uploading"
+
+func TestContent(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: POST /media/r0/upload can create an upload
+	t.Run("POST /media/r0/upload can create an upload", func(t *testing.T) {
+
+		gotUrl := ""
+		queryParams := url.Values{}
+		queryParams.Set("access_token", authedClient.AccessToken)
```

Comment:
This shouldn't be necessary. Just use an authed client as always. The perl test didn't because the framework wouldn't let you if you weren't sending JSON.